lib/Maintenance/Executor.php

```diff
@@ -44,69 +46,83 @@ final class Executor implements ExecutorInterface
      */
     private $lockFactory = null;
 
-    /**
-     * @param string $pidFileName
-     * @param LoggerInterface $logger
-     * @param LockFactory $lockFactory
-     */
-    public function __construct(string $pidFileName, LoggerInterface $logger, LockFactory $lockFactory)
+    public function __construct(
+        string $pidFileName,
+        LoggerInterface $logger,
+        LockFactory $lockFactory,
+        private MessageBusInterface $messageBus
+    )
     {
         $this->pidFileName = $pidFileName;
         $this->logger = $logger;
         $this->lockFactory = $lockFactory;
     }
 
+    public function executeTask(string $name, bool $force = false)
+    {
+        if (!in_array($name, $this->getTaskNames(), true)) {
+            throw new \Exception(sprintf('Task with name "%s" not found', $name));
+        }
+
+        $task = $this->tasks[$name];
+        $lock = $this->lockFactory->createLock('maintenance-' . $name, 86400);
+
+        if (!$lock->acquire() && !$force) {
```

Comment:
Shouldn't we leave the `$force` check in `executeMaintenance()`. There's no need to add them to the queue if not necessary, right?

Reply:
the task could be added to the queue from other places too... so I thought the TaskHandler/executeTask should be responsible for it